lib/modules/manager/gradle/extract/catalog.ts

```diff
@@ -205,7 +207,7 @@ function extractDependency({
     };
   }
   const versionRef = isVersionPointer(descriptor.version)
-    ? descriptor.version.ref
+    ? descriptor.version.ref.replaceAll('.', '-').replaceAll('_', '-')
```

Comment:
```suggestion
    ? descriptor.version.ref.replace(/[._]/g, '-')
```
https://jsben.ch/KY0Cv

Reply:
@Churro 
do i need to use `regEx(/[._]/g`) library with this?

Reply:
make sense 👍